app/models/service_template_ansible_playbook.rb

```diff
@@ -156,11 +156,17 @@ def update_catalog_item(options, auth_user = nil)
 
   def destroy
     auth_user = User.current_userid || 'system'
+    raise 'Attached retirement configurations do not allow the service template to be deleted.' if retirement_defined?
```

Comment:
@bzwei - In the case where  I would need to return an error - do we append the message to an `errors` hash? - Not sure the pattern we use in our app for that callback.